### PR TITLE
Activation checkpointing in FSDP without boilerplate

### LINKED ISSUE #15825: Add support for activation checkpointing in FSDP without boilerplate
### Motivation

Adding activation checkpointing in FSDP is easy but requires to write some boilerplate code.
See the tutorial here: https://www.youtube.com/watch?v=5B4d0FuxSQc

### Pitch

Expose an argument in FSDP to which the user can pass their transformer block.

Before:

```py
lite = LightningLite(strategy=FSDPStrategy(...))
model = lite.setup_model(...)

from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
    CheckpointImpl, apply_activation_checkpointing, checkpoint_wrapper
)
check_fn = lambda submodule: isinstance(submodule, MyTransformerBlock)
wrapper = functools.partial(
    checkpoint_wrapper,
    offload_to_cpu=False,
    checkpoint_impl=CheckpointImpl.NO_REENTRANT,
)
apply_activation_checkpointing(
    model, checkpoint_wrapper_fn=wrapper, check_fn=check_fn
)

```

After:
```py
lite = LightningLite(strategy=FSDPStrategy(..., activation_checkpointing=MyTransformerBlock))
```

### Alternatives

😡 

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

cc @borda @carmocca @justusschock @awaelchli

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #15825

TODO: 
- [x] add the same for the PL implementation
- [x] tests
- [x] version check  
- [x] docs 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: src/lightning_lite/strategies/fsdp.py [hunk 1]
```diff
@@ -291,6 +309,22 @@ def _set_world_ranks(self) -> None:
...
+    from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
+        apply_activation_checkpointing,
+        checkpoint_wrapper,
+        CheckpointImpl,
+    )
+
+    check_fn = lambda submodule: isinstance(submodule, tuple(layers))
+    wrapper = functools.partial(
+        checkpoint_wrapper,
+        offload_to_cpu=False,
```
Comment (lantiga) [inline src/lightning_lite/strategies/fsdp.py:322]: What would be the disadvantage of leaving CPU offloading as an option?

**Review by lantiga — APPROVED** [review]

File: src/lightning_lite/strategies/fsdp.py (hunk 1 above)
Comment (awaelchli) [reply src/lightning_lite/strategies/fsdp.py:322]: I just explicitly wrote it to signal that this is the default setting in FSDP. This is specific to version 1.13. The argument was already removed on master (1.14 nightly) and so I wouldn't expose it now unless their API becomes stable.

**Review by justusschock — APPROVED** [review]
